[Yin Ruoyan] iP

ruoyann opened the pull request:
# DUKE BOT
> "I forgot", never again!

Duke helps you to keep track of **all** your tasks. It is
* Easy to use
* No lags *at all*

What you need to do
1. download it [here] (https://github.com/ruoyann/ip/releases/tag/A-Jar)
2. run `java -jar <insert file path where the above file is>`
3. that's all!  :rofl:

Features
- [x] Able to add tasks such as todo, event, deadline
- [x] Able to mark tasks as done, delete tasks, and find tasks
- [ ] Graddle, GUI

Here's the main method: 

```java
public static void main(String[] args) {
        new Duke(FILE_PATH).run();
    }
```

Commit:
Add Gradle support

ruoyann committed:
Level-1

ruoyann committed:
Level-2

ruoyann committed:
Level-3

ruoyann committed:
Level-4

ruoyann committed:
Level-4

ruoyann committed:
A-TextUiTesting

ruoyann committed:
A-Exceptions

ruoyann committed:
Level-6

ruoyann committed:
more exceptions added

ruoyann committed:
Level 7

ruoyann committed:
Level 8

ruoyann committed:
Merge branch 'branch-Level-7'

ruoyann committed:
Merge branch 'branch-Level-8'

ruoyann committed:
A-MoreOOP

ruoyann committed:
A-MoreOOP bug fix

ruoyann committed:
A-Packages

ruoyann committed:
A-JUnit

ruoyann committed:
A-Jar

ruoyann committed:
A-JavaDoc

ruoyann committed:
A-CodingStandard

ruoyann committed:
Level-9

ruoyann committed:
Merge tag 'A-JavaDoc'

ruoyann committed:
Conflicts resolution

ruoyann committed:
Conflict resolution for merging level 9

ruoyann committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ruoyann committed:
A-Gradle

ruoyann committed:
Merge branch 'branch-A-Gradle'

ruoyann committed:
bug: Fixed bug for gradle

ruoyann committed:
Merge branch 'branch-A-Gradle'

ruoyann committed:
code cleanup

aikenwx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import duke.task.TaskList;
```
Should check ordering for imports.

aikenwx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+import duke.ui.Ui;
+import duke.exception.DukeException;
+
+import java.io.*;
```
.* form of imports should be avoided.

aikenwx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    private static final String FILE_PATH = String.valueOf(Paths.get(System.getProperty("user.home"), "data", "dukeFile.txt"));
```
Line is too long (longer than 120 chars)

aikenwx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+            System.out.println("Caught the exception");
```
Wrong indentation level

aikenwx commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ * The class for a user command that adds a task to the list
+ */
+public class AddCommand extends Command{
```
'{' is not preceded with white space

aikenwx commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,25 @@
+    /**
+     * Method that executes the parsing of user input and the message
+     * to print to the command line
+     * @param taskList the current list of tasks
```
taskList is an unused param tag.

aikenwx commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,147 @@
+        }
+        int byIndex = input.indexOf("/");
+        String by = input.substring(byIndex+4);
+        String taskDesc = input.substring(9, byIndex-1);
```
Add spaces between '-' operator

aikenwx commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,129 @@
+                currLine += ((Deadline)t).getBy();
+            } else if (t instanceof Event) {
+                currLine += " | ";
+                currLine += ((Event)t).getAt();
```
Typecast should be followed by whitespace.

aikenwx reviewed (commented):
Might want to consider setting up CheckStyle to find coding standard violations.

ruoyann committed:
Coding Standard Violations

ruoyann committed:
branch error fix

ruoyann committed:
Level-10

ruoyann committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/resources/view/MainWindow.fxml

ruoyann committed:
Fix bugs

bentanjunrong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,90 @@
+    public String getUiWelcome() {
+        return this.ui.showWelcome();
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,90 @@
+     */
+    String getResponse(String input) {
+        String res = "";
+//        ui.showWelcome();
```
Minor (Coding Standard): Remember to remove unused code comments!

bentanjunrong commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,12 @@
+    public static void main(String[] args) {
+        Application.launch(Main.class, args);
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,54 @@
+        );
+        userInput.clear();
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,40 @@
+        tasks.addTask(task);
+        return ui.addTask(task, tasks);
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,10 @@
+    public DukeException(String message) {
+        super(message);
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,38 @@
+    public String toString() {
+       return ("[" + this.getStatusIcon() + "] " + this.description);
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,45 @@
+        }
+        return new TaskList(foundTasks);
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,27 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/resources/view/MainWindow.fxml:
```diff
@@ -0,0 +1,19 @@
+            </content>
+        </ScrollPane>
+    </children>
+</AnchorPane>
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/resources/view/DialogBox.fxml:
```diff
@@ -0,0 +1,16 @@
+    <padding>
+        <Insets bottom="15.0" left="5.0" right="5.0" top="15.0" />
+    </padding>
+</fx:root>
```
Minor (Coding Standard): Missing newline at the end of the file!

bentanjunrong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,154 @@
+            String target = parseFind(input);
+            return new FindCommand(target);
+        } else {
+            if (!input.startsWith("todo") && !input.startsWith("deadline") && !input.startsWith("event")) {
```
This is double checking for input validity! Would be cleaner to check for invalid command input at the end of this else block e.g. if else statements checking if it starts with "todo"/"event"/"deadline", and one last else that throws a DukeException otherwise.

bentanjunrong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,154 @@
+     * @throws DukeException if the user's input is in an wrong format
+     */
+    public static String parseFind(String input) throws DukeException {
+        if (input.length() == 4) {
```
I think it's advisable not to check for validity using String length in this case, especially if you're not sanitising the input using .trim() . For example, if the input was "todo " with whitespaces at the end, it should be considered an invalid input, but I believe your code does not catch it.

bentanjunrong reviewed (commented):
Overall great job! OOP principles were followed well, and the code itself is written in a logical and clean way. Nice 👍

ruoyann committed:
There are no assertions at all

Assertions are useful to prove the correctness of the code

Assertion about input length should be added

To show correctness of code

ruoyann committed:
Fix Code Quality

Code does not follow code quality guidelines

Following code quality makes code more readable

Dead code should be deleted, variables should be made final if necessary

For code to be more easily understood by readers

ruoyann committed:
bug: Fix truncated dialog box

ruoyann committed:
Merge pull request #1 from ruoyann/branch-A-Assertions

Add Assert statements

ruoyann committed:
Merge pull request #2 from ruoyann/branch-A-CodeQuality

Code Quality Branch

ruoyann committed:
Add C-Priority extension

ruoyann committed:
Fix bug where storage file was not properly updated

ruoyann committed:
Add UI screenshot, change DaDuke and DaUser pictures

ruoyann committed:
Add user guide

ruoyann committed:
Set theme jekyll-theme-minimal

ruoyann committed:
Change file path to be relative

ruoyann committed:
Merge branch 'master' of https://github.com/ruoyann/ip

ruoyann committed:
Throw exception when priority not added

ruoyann committed:
Edit readme typo

ruoyann committed:
Add more exceptions

ruoyann committed:
Exit platform with 'byeCommand'

ruoyann committed:
Remove platform.close() when 'bye' command is entered
